lib/babe/babe.go

```diff
@@ -263,7 +263,9 @@ func (b *Service) Stop() error {
 // Authorities returns the current BABE authorities
 func (b *Service) Authorities() []types.Authority {
 	auths := make([]types.Authority, len(b.epochHandler.epochData.authorities))
-	copy(auths, b.epochHandler.epochData.authorities)
+	for i, auth := range b.epochHandler.epochData.authorities {
+		auths[i] = *(auth.DeepCopy())
```

Comment:
nit do you need the parentheses? I think this would work too

```suggestion
		auths[i] = *auth.DeepCopy()
```